tempesta_fw/http_limits.c

```diff
@@ -538,6 +538,25 @@ frang_http_methods(const TfwHttpReq *req, FrangAcc *ra, unsigned long m_mask)
 	return TFW_PASS;
 }
 
+static int
+frang_http_methods_override(const TfwHttpReq *req, FrangAcc *ra,
+			    FrangVhostCfg *f_cfg)
+{
+	unsigned long mbit = (1UL << req->method_override);
+
+	if (!req->method_override)
+		return TFW_PASS;
+	if (!f_cfg->http_method_override
+	    || (f_cfg->http_methods_mask && !(f_cfg->http_methods_mask & mbit)))
```

Comment:
The second part of the expression is equal to `!(f_cfg->http_methods_mask & mbit)`, i.e. `f_cfg->http_methods_mask` is unnecessary

Reply:
The `f_cfg->http_methods_mask` part in condition is actually required. If the `http_methods` directive under `frang_limits` section isn't declared, then all methods are allowed, and `f_cfg->http_methods_mask`  is `0`.  That's not great and it seems like the mask should be `0xffffffff` by default, but this option follows frang global configuration rules: if something is disabled it's zero.

Thus we have three options how to deal with overridden method:
- block a request if overriding is prohibited.,
- pass a request if overriding is allowed and method restriction is not used.
- block a request if overridden method is prohibited, pass the request otherwise.